store/views.py

```diff
@@ -1,196 +1,231 @@
-from django.shortcuts import render , get_object_or_404 , redirect
-from django.views import View
-from django.views.generic.detail import DetailView 
-from django.views.generic import TemplateView , ListView 
-from .models import Book , Category , User , Cart , CartItem, Order, OrderItem, Profile
-from django.db.models import Q
-from . forms import RegistrationForm, ProfileForm, BookForm
-from django.contrib.auth import login , authenticate , logout
 from django.contrib import messages
-from django.contrib.auth.forms import AuthenticationForm
+from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.forms import AuthenticationForm
+from django.db.models import Q
+from django.shortcuts import get_object_or_404, redirect, render
 from django.utils import timezone
+from django.views import View
+from django.views.generic import ListView, TemplateView
+from django.views.generic.detail import DetailView
+
+from .forms import BookForm, ProfileForm, RegistrationForm
+from .models import (Book, Cart, CartItem, Category, Order, OrderItem, Profile,
+                     User)
 
 # Create your views here.
 
 
 # views.py
 
+
 class HomeView(TemplateView):
-    template_name = 'home.html'
-  
-    def get(self , request):
+    template_name = "home.html"
+
+    def get(self, request):
         categories = Category.objects.all()
         books = Book.objects.order_by("created")
-        context = {"categories" : categories , "books" : books}
-        return render(request , self.template_name , context)
+        user_type = (
+            request.user.user_type if request.user.is_authenticated else None
+        )
+        context = {
+            "categories": categories,
+            "books": books,
+            "user_type": user_type,
+        }
+        return render(request, self.template_name, context)
 
 
-class CategoryDetailView(DetailView) :
+class CategoryDetailView(DetailView):
```

Comment:
@MansiS117 I want to know if you are using detailview then why did you override get method here?

Reply:
sir, i am overriding the get method here because i wanted to include an additional context data i.e book_count so that i could render no. of books related to a particular category

Reply:
@MansiS117 I think you can do something better than overriding get. please do some research.

Reply:
okay sir

Reply:
Overriding the get_context_data method instead of the get method, would it be a correct approach sir?

Reply:
okay sir, i will implement it

Reply:
sir, implemented the suggested changes